### PR TITLE
Rustup

### PR DESCRIPTION
**Pull request by mati865**
All tests passed locally.

`Reveal::all` is now [default behaviour](https://github.com/rust-lang/rust/commit/6d0f9319dfd4a115f64fd5e00be3749da5eda8bc).

Fixes https://github.com/rust-lang-nursery/rust-clippy/issues/2528

### CODE REVIEW

**Comment by oli-obk** [comment]
@Manishearth @llogiq can you publish a new version? I'm on mobile until tomorrow

**Comment by mati865** [comment]
Merge broken 1 more test. Brb 5 min

**Comment by oli-obk** [comment]
That last update is wrong. You need to cargo clean first

File: tests/ui/useless_attribute.stderr [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
   |
-5 | #[allow(dead_code, unused_extern_crates)]
-  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: if you just forgot a `!`, use: `#![allow(dead_code, unused_extern_crates)]`
+9 | extern crate clippy_lints;
+  | ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
-  = note: `-D useless-attribute` implied by `-D warnings`
+  = note: candidates:
+          crate `clippy_lints`: /home/mateusz/temp/rust-clippy/target/debug/deps/libclippy_lints-9e310b886f99a60d.rlib
+          crate `clippy_lints`: /home/mateusz/temp/rust-clippy/target/debug/deps/libclippy_lints-72c784bbede1859c.rlib
```
Comment (mati865) [inline tests/ui/useless_attribute.stderr]: Looks like `tests/ui/update-references.sh` misbehaved. I'm gonna cleanup this mess

**Comment by mati865** [comment]
Okay this time cleaned up target and updated tests again.
Lints pass:
```
test result: ok. 183 passed; 0 failed; 1 ignored; 0 measured; 0 filtered out

test compile_test ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```

But dogfood woke up and panicked after merging master:
```
thread 'rustc' panicked at 'Box<Any>', librustc_errors/lib.rs:540:9
note: Run with `RUST_BACKTRACE=1` for a backtrace.

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.26.0-nightly (521d91c6b 2018-03-14) running on x86_64-unknown-linux-gnu

note: compiler flags: -C debuginfo=2 -C incremental --crate-type lib

note: some of the compiler flags provided by cargo are hidden

error: Could not compile `clippy_lints`.

To learn more, run the command again with --verbose.

thread 'dogfood' panicked at 'assertion failed: output.status.success()', tests/dogfood.rs:21:9
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

**Comment by mati865** [comment]
I don't know if this ICE is relevant, @oli-obk you might want to look into it when you are back.
<details>
  <summary>Click here for ICE</summary>
  <p>

```
thread 'rustc' panicked at 'Box<Any>', librustc_errors/lib.rs:540:9
stack backtrace:
   0:     0x7fbc5586f9a3 - std::sys::unix::backtrace::tracing::imp::unwind_backtrace::h6b1aa521439b5a8f
                               at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1:     0x7fbc55866854 - std::sys_common::backtrace::_print::hf0c55570aef1b27d
                               at libstd/sys_common/backtrace.rs:71
   2:     0x7fbc5586c23d - std::panicking::default_hook::{{closure}}::h488bb196ba72cdaa
                               at libstd/sys_common/backtrace.rs:59
                               at libstd/panicking.rs:207
   3:     0x7fbc5586bf39 - std::panicking::default_hook::hef52c65948961f87
                               at libstd/panicking.rs:223
   4:     0x7fbc567bbf8d - core::ops::function::Fn::call::h7157cbafb9ba04dd
   5:     0x7fbc5586c6d9 - std::panicking::rust_panic_with_hook::h8de03ed35cb0aca6
                               at libstd/panicking.rs:403
   6:     0x7fbc55f9d9ea - std::panicking::begin_panic::h05829b503f9a2319
   7:     0x7fbc55fbe3c5 - rustc_errors::Handler::bug::h74f7209403457daf
   8:     0x7fbc56ac2aff - rustc::session::opt_span_bug_fmt::{{closure}}::h6cfbec07311b4575
   9:     0x7fbc56b7f1e5 - rustc::ty::context::tls::with_opt::{{closure}}::h6da43cb2f4e4a601
  10:     0x7fbc567b480f - <std::thread::local::LocalKey<T>>::try_with::ha72882829f133149
  11:     0x7fbc567b4756 - <std::thread::local::LocalKey<T>>::with::hb8bed0f7763ac92e
  12:     0x7fbc56b7e6b6 - rustc::ty::context::tls::with::hba5ebd0b698effa2
  13:     0x7fbc56b7f197 - rustc::ty::context::tls::with_opt::hbed81867dc059708
  14:     0x7fbc56ac2a04 - rustc::session::opt_span_bug_fmt::h46e1d5bb7b8d6d36
  15:     0x7fbc56ac2976 - rustc::session::bug_fmt::h92c4760bc71e1d9d
  16:     0x7fbc56b792aa - rustc::ty::context::InterpretInterner::cache::h60a0a659f94be72e
  17:     0x7fbc56b5702e - <rustc::ty::maps::on_disk_cache::CacheDecoder<'a, 'tcx, 'x> as serialize::serialize::SpecializedDecoder<rustc::mir::interpret::AllocId>>::specialized_decode::h2c9057ecf5f0d384
  18:     0x7fbc56b56e62 - <rustc::ty::maps::on_disk_cache::CacheDecoder<'a, 'tcx, 'x> as serialize::serialize::SpecializedDecoder<rustc::mir::interpret::AllocId>>::specialized_decode::h2c9057ecf5f0d384
  19:     0x7fbc5692e378 - serialize::serialize::Decoder::read_map::h060680f8a8efb76b
  20:     0x7fbc56bc8cda - <rustc::mir::interpret::Allocation as serialize::serialize::Decodable>::decode::{{closure}}::h36e7447e2bddf031
  21:     0x7fbc56b56b88 - <rustc::ty::maps::on_disk_cache::CacheDecoder<'a, 'tcx, 'x> as serialize::serialize::SpecializedDecoder<rustc::mir::interpret::AllocId>>::specialized_decode::h2c9057ecf5f0d384
  22:     0x7fbc5692e378 - serialize::serialize::Decoder::read_map::h060680f8a8efb76b
  23:     0x7fbc56bc8cda - <rustc::mir::interpret::Allocation as serialize::serialize::Decodable>::decode::{{closure}}::h36e7447e2bddf031
  24:     0x7fbc56b56b88 - <rustc::ty::maps::on_disk_cache::CacheDecoder<'a, 'tcx, 'x> as serialize::serialize::SpecializedDecoder<rustc::mir::interpret::AllocId>>::specialized_decode::h2c9057ecf5f0d384
  25:     0x7fbc56bc89d6 - <rustc::mir::interpret::MemoryPointer as serialize::serialize::Decodable>::decode::{{closure}}::hfebdd15481f4b2cf
  26:     0x7fbc5692c509 - serialize::serialize::Decoder::read_enum_variant::h77a01fe28278938b
  27:     0x7fbc56bc858f - <rustc::mir::interpret::value::Pointer as serialize::serialize::Decodable>::decode::{{closure}}::h4c84becedf337564
  28:     0x7fbc5692b53b - serialize::serialize::Decoder::read_enum_variant::h3e2a115d47f0d0a0
  29:     0x7fbc5692c95f - serialize::serialize::Decoder::read_enum_variant::he3af156c43481578
  30:     0x7fbc56c5b94a - rustc::ty::maps::<impl rustc::ty::maps::queries::const_eval<'tcx>>::try_get::h0d12b3ad6c2f7468
  31:     0x7fbc56d50a6a - rustc::ty::maps::TyCtxtAt::const_eval::hede1d2b35ef1a476
  32:     0x7fbc56d4d084 - rustc::ty::maps::<impl rustc::ty::context::TyCtxt<'a, 'tcx, 'lcx>>::const_eval::h89fff34c55c83ffb
  33:     0x7fbc52eb59a9 - rustc_mir::monomorphize::collector::collect_items_rec::h7a76a985655ace38
  34:     0x7fbc52eb654e - rustc_mir::monomorphize::collector::collect_items_rec::h7a76a985655ace38
  35:     0x7fbc52eb654e - rustc_mir::monomorphize::collector::collect_items_rec::h7a76a985655ace38
  36:     0x7fbc52eb5352 - rustc_mir::monomorphize::collector::collect_crate_mono_items::he4856bb7b5fad57f
  37:     0x7fbc4c8fc86b - rustc_trans::base::collect_and_partition_translation_items::h3731a3e7942a5609
  38:     0x7fbc56d2652d - rustc::ty::maps::<impl rustc::ty::maps::queries::collect_and_partition_translation_items<'tcx>>::try_get::hfcd459a65f4f8db5
  39:     0x7fbc56d554f5 - rustc::ty::maps::TyCtxtAt::collect_and_partition_translation_items::h31a7a1b21f210409
  40:     0x7fbc56d4e34e - rustc::ty::maps::<impl rustc::ty::context::TyCtxt<'a, 'tcx, 'lcx>>::collect_and_partition_translation_items::hf7d4ba6667bf6992
  41:     0x7fbc4c8f8273 - rustc_trans::base::trans_crate::h2df22e25d73e994d
  42:     0x7fbc4c94fca7 - <rustc_trans::LlvmTransCrate as rustc_trans_utils::trans_crate::TransCrate>::trans_crate::ha2cc4dd8fe54beb7
  43:     0x7fbc577878fc - rustc_driver::driver::phase_4_translate_to_llvm::h9803bf4339df30ed
  44:     0x7fbc577802fe - rustc_driver::driver::compile_input::{{closure}}::h9c1a88c91165bccc
  45:     0x7fbc5771a4d7 - rustc::ty::context::TyCtxt::create_and_enter::hb54e14a06694259e
  46:     0x7fbc5777e321 - rustc_driver::driver::compile_input::h6560736e056a863b
  47:     0x7fbc5779b435 - rustc_driver::run_compiler::h060b34b2732dceee
  48:     0x561b184fd958 - clippy_driver::main::{{closure}}::h0724b9e487aa175c
                               at src/driver.rs:197
  49:     0x561b18524230 - rustc_driver::run::{{closure}}::h4b35ea84998ac14d
                               at /checkout/src/librustc_driver/lib.rs:172
  50:     0x561b185250e6 - rustc_driver::monitor::{{closure}}::h9bc0435473c87082
                               at /checkout/src/librustc_driver/lib.rs:1540
  51:     0x561b1852e892 - std::sys_common::backtrace::__rust_begin_short_backtrace::ha2616346ebd80e72
                               at /checkout/src/libstd/sys_common/backtrace.rs:136
  52:     0x561b184f47f6 - std::thread::Builder::spawn::{{closure}}::{{closure}}::hea6b92d83df38f8b
                               at /checkout/src/libstd/thread/mod.rs:406
  53:     0x561b1851ccc2 - <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::hd84421b912b1c143
                               at /checkout/src/libstd/panic.rs:296
  54:     0x561b1850dd5c - std::panicking::try::do_call::h0eec6ce162ea5c38
                               at /checkout/src/libstd/panicking.rs:306
  55:     0x7fbc5588553e - __rust_maybe_catch_panic
                               at libpanic_unwind/lib.rs:102
  56:     0x561b1850dc68 - std::panicking::try::h040e0d0056d526a3
                               at /checkout/src/libstd/panicking.rs:285
  57:     0x561b1851cd09 - std::panic::catch_unwind::h7f5ace7f41daa4f3
                               at /checkout/src/libstd/panic.rs:361
  58:     0x561b184f4654 - std::thread::Builder::spawn::{{closure}}::hd4e38391f8847660
                               at /checkout/src/libstd/thread/mod.rs:405
  59:     0x561b184f490f - <F as alloc::boxed::FnBox<A>>::call_box::hff280fa192bdd238
                               at /checkout/src/liballoc/boxed.rs:783
  60:     0x7fbc5587c79b - std::sys::unix::thread::Thread::new::thread_start::h3d29264a23eaa0fc
                               at /checkout/src/liballoc/boxed.rs:793
                               at libstd/sys_common/thread.rs:24
                               at libstd/sys/unix/thread.rs:90
  61:     0x7fbc555e508b - start_thread
  62:     0x7fbc55105e7e - __GI___clone
  63:                0x0 - <unknown>

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.26.0-nightly (521d91c6b 2018-03-14) running on x86_64-unknown-linux-gnu

note: compiler flags: -C debuginfo=2 -C incremental --crate-type lib

note: some of the compiler flags provided by cargo are hidden

error: Could not compile `clippy_lints`.

To learn more, run the command again with --verbose.

thread 'dogfood' panicked at 'assertion failed: output.status.success()', tests/dogfood.rs:21:9
stack backtrace:
   0:     0x55b2d5aa3093 - std::sys::unix::backtrace::tracing::imp::unwind_backtrace::h6b1aa521439b5a8f
                               at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1:     0x55b2d5a9d6a4 - std::sys_common::backtrace::_print::hf0c55570aef1b27d
                               at libstd/sys_common/backtrace.rs:71
   2:     0x55b2d5aa20cd - std::panicking::default_hook::{{closure}}::h488bb196ba72cdaa
                               at libstd/sys_common/backtrace.rs:59
                               at libstd/panicking.rs:207
   3:     0x55b2d5aa1d62 - std::panicking::default_hook::hef52c65948961f87
                               at libstd/panicking.rs:217
   4:     0x55b2d5aa252f - std::panicking::rust_panic_with_hook::h8de03ed35cb0aca6
                               at libstd/panicking.rs:402
   5:     0x55b2d5a52973 - std::panicking::begin_panic::h9e851d166d31f46d
                               at /checkout/src/libstd/panicking.rs:365
   6:     0x55b2d5a51387 - dogfood::dogfood::hde68387b549eb795
                               at tests/dogfood.rs:21
   7:     0x55b2d5a52530 - dogfood::__test::TESTS::{{closure}}::h3e58f8854861358b
                               at tests/dogfood.rs:2
   8:     0x55b2d5a51850 - core::ops::function::FnOnce::call_once::h8753663987ba6ec5
                               at /checkout/src/libcore/ops/function.rs:223
   9:     0x55b2d5a60b61 - <F as alloc::boxed::FnBox<A>>::call_box::hd5afb48f7fa14d1b
                               at libtest/lib.rs:1477
                               at /checkout/src/libcore/ops/function.rs:223
                               at /checkout/src/liballoc/boxed.rs:783
  10:     0x55b2d5ab062e - __rust_maybe_catch_panic
                               at libpanic_unwind/lib.rs:102
  11:     0x55b2d5a585b3 - std::sys_common::backtrace::__rust_begin_short_backtrace::h32b834d0e4f24c09
                               at /checkout/src/libstd/panicking.rs:285
                               at /checkout/src/libstd/panic.rs:361
                               at libtest/lib.rs:1429
                               at /checkout/src/libstd/sys_common/backtrace.rs:136
  12:     0x55b2d5a595c2 - std::panicking::try::do_call::h88ad6e1120bff6cb
                               at /checkout/src/libstd/thread/mod.rs:406
                               at /checkout/src/libstd/panic.rs:296
                               at /checkout/src/libstd/panicking.rs:306
  13:     0x55b2d5ab062e - __rust_maybe_catch_panic
                               at libpanic_unwind/lib.rs:102
  14:     0x55b2d5a60972 - <F as alloc::boxed::FnBox<A>>::call_box::h09683460452de998
                               at /checkout/src/libstd/panicking.rs:285
                               at /checkout/src/libstd/panic.rs:361
                               at /checkout/src/libstd/thread/mod.rs:405
                               at /checkout/src/liballoc/boxed.rs:783
  15:     0x55b2d5aa857b - std::sys::unix::thread::Thread::new::thread_start::h3d29264a23eaa0fc
                               at /checkout/src/liballoc/boxed.rs:793
                               at libstd/sys_common/thread.rs:24
                               at libstd/sys/unix/thread.rs:90
  16:     0x7fbb1ea5e08b - start_thread
  17:     0x7fbb1e57ee7e - __GI___clone
  18:                0x0 - <unknown>


failures:
    dogfood
```

</p></details>


Anyway clippy works fine and Travis passes (AppVeyor hanged on the other PR).
Sorry for that mess at the beginning.

**Comment by oli-obk** [comment]
Yea... looks very much related. I gotta fix this in rustc.

**Comment by oli-obk** [comment]
Huh? How is Travis passing?

**Comment by mati865** [comment]
I tried to apply dogfood suggestions but apparently some of them break the code. That's how it looks: https://hastebin.com/omuxebegid

Pushing only fix for warnings I missed.

**Comment by oli-obk** [comment]
Thx. I'll merge for now so git users can get it

**Comment by mati865** [comment]
Thanks.
I actually managed to get rid of that ICE by fixing the code according to the suggestions. I need to finish it and I'll open PR later.

**Pull request merged**